platform/linux-generic/m4/odp_crypto.m4

```diff
@@ -0,0 +1,33 @@
+# ODP_CRYPTO
+# ----------
+# Select default crypto implementation
+AC_ARG_ENABLE([crypto-default],
+             [AS_HELP_STRING([--enable-crypto-default],
+                             [choose default crypto implementation [openssl/armv8crypto/null] (linux-generic)])],
+             [default_crypto=$enableval], [default_crypto=openssl])
+
+AS_IF([test "x$default_crypto" != "xopenssl" -a "x$default_crypto" != "xarmv8crypto" -a "x$default_crypto" != "xnull"],
+      [AC_MSG_ERROR([Invalid crypto implementation name])])
```

Comment:
Unfortunately, I wasn't a part of the conversation when these changes were being discussed so I don't know the idea behind it. But, I do understand the confusion that you're talking about. 
The autoconf option, `--enable-crypto-default` was added to allow to choose between different crypto implementations (OpenSSL/Armv8crypto/Null). If we simply build with `./configure`, it chooses OpenSSL by default. It will be a good idea to change the name to `--with-crypto` rather than `--enable-crypto-default` so that it is clearly understood that the option is for choosing different crypto implementations and not enabling/disabling crypto itself. 
Let me know what you think. Thanks!

Reply:
I agree that the argument name could be improved. I guess the `--enable-crypto-default` name was just copied from the existing `--enable-scheduler-default` option.

`--with-crypto` sounds like a good name. And as Janne mentioned, `yes`/`no` values should also be accepted.